osu.Game.Rulesets.Taiko/Mods/TaikoModHidden.cs

```diff
@@ -2,34 +2,43 @@
 // See the LICENCE file in the repository root for full licence text.
 
 using osu.Framework.Graphics;
-using osu.Game.Beatmaps;
-using osu.Game.Beatmaps.ControlPoints;
 using osu.Game.Rulesets.Mods;
 using osu.Game.Rulesets.Objects;
 using osu.Game.Rulesets.Objects.Drawables;
 using osu.Game.Rulesets.Scoring;
+using osu.Game.Rulesets.Taiko.Objects;
 using osu.Game.Rulesets.Taiko.Objects.Drawables;
+using osu.Game.Rulesets.Taiko.UI;
+using osu.Game.Rulesets.UI;
 
 namespace osu.Game.Rulesets.Taiko.Mods
 {
-    public class TaikoModHidden : ModHidden
+    public class TaikoModHidden : ModHidden, IApplicableToDrawableRuleset<TaikoHitObject>
     {
         public override string Description => @"Beats fade out before you hit them!";
         public override double ScoreMultiplier => 1.06;
 
-        private ControlPointInfo controlPointInfo;
+        /// <summary>
+        /// How far away from the hit target should hitobjects start to fade out.
+        /// Range: [0, 1]
+        /// </summary>
+        private const float fade_out_start_time = 0.6f;
 
-        protected override void ApplyIncreasedVisibilityState(DrawableHitObject hitObject, ArmedState state)
+        /// <summary>
+        /// How long hitobjects take to fade out, in terms of the scrolling length.
+        /// Range: [0, 1]
+        /// </summary>
+        private const float fade_out_duration = 0.3f;
+
+        private DrawableTaikoRuleset drawableRuleset;
+
+        public void ApplyToDrawableRuleset(DrawableRuleset<TaikoHitObject> drawableRuleset)
         {
-            ApplyNormalVisibilityState(hitObject, state);
```

Comment:
I've also grouped this change in here since it's just a removal of one line. I see no reason for increased visibility to still hide the first hitobject. If players don't want this then they can just disable the setting.

Reply:
catch's hidden mod also does this, so if we're removing then it should probably get removed everywhere.

Reply:
Can agree with removing it everywhere, would prefer a separate PR just so there's visibility in the changelog etc.

Reply:
I'll remove it from this one too then.